docs/sources/auth/saml.md

```diff
@@ -0,0 +1,136 @@
+| `private_key|_path`       | Yes      | Base64-encoded string or Path for the SP private key                           |         |
+| `idp_metadata|_path`      | Yes      | Base64-encoded string or Path for the IdP SAML metadata XML                    |         |
+| `max_issue_delay`         | No       | Duration, since the IdP issued a response and the SP is allowed to process it. | `90s`   |
+| `metadata_valid_duration` | No       | Duration, for how long the SP's metadata should be valid.                      | `48h`   |
+
+## `certificate` and `private_key`
+
+The SAML SSO standard uses asymmetric encryption to exchange information between the SP (Grafana) and the IdP. To perform such encryption, we need a public part and a private part. In this case, the X.509 certificate provides the public part, while the private key provides the private part.
+
+Grafana supports two ways of specifying both the `certificate` and `private_key`. Without a suffix (e.g. `certificate=`), the configuration assumes you've supplied the base64-encoded file contents. However, if specified with the `_path` suffix (e.g. `certificate_path=`) we treat it as a file path and attempt to read the file from the file system.
+
+## `idp_metadata`
+Expanding on the above, we'll also need the public part from our IdP for message verification. The SAML IdP metadata XML tells us where and how we should exchange the user information.
+
+Currently, we support two ways of specifying the IdP metadata. Without a suffix `idp_metadata=` we'll assume base64-encoded XML file contents, and with the `_path` suffix we treat it as a file path and attempt to read the file from the file system.
+
+## `max_issue_delay`
+
+To prevent SAML response replay attacks and internal clock skews between the SP (Grafana) and the IdP. We can set a maximum amount of time between the IdP issuing a response and the SP (Grafana) processing it.
+
+The configuration options is specified as a duration e.g. `max_issue_delay= 90s` or `max_issue_delay= 1h`
+
+## `metadata_valid_duration`
+
+As an SP, our metadata is likely to expire at some point, e.g. due to a certificate rotation or change of location binding. Grafana allows you to specify for how long should the metadata be valid. Leveraging the standard's `validUntil` field, we can tell consumers until when our metadata is going to be valid. The duration is computed by adding the duration to the current time.
+ 
+The configuration option is specified as a duration e.g. `metadata_valid_duration = 48h`
+
+# Identity Provider (IdP) registration
+
+Finally, for the SAML integration to work correctly, we need to make the IdP aware of the SP. You need to provide the IdP with the SP's metadata. 
+
+Grafana provides an endpoint for such at `grafana_url.com/saml/metadata`. You can either download the metadata and upload it manually, or make the IdP request it directly from the endpoint.
```

Comment:
With the urls, it seems, we are tend to just write relative path without the domain. So instead of `grafana_url.com/saml/metadata` just `/saml/metadata`

Reply:
Thanks! Was wondering that given I couldn't find an example.